README.md

```diff
@@ -0,0 +1,135 @@
+# zm-build
+
+## Introduction
+
+This repository contains the build script and supporting files required to create a [FOSS](https://en.wikipedia.org/wiki/Free_and_open-source_software) build of the [Zimbra Collaboration Suite](https://www.zimbra.com/). 
+
+## Overview
+
+* `build.pl` - Invoke this script to produce a build.  See the *Building* section 
+  below for an example.
+* `instructions/`
+    * `FOSS_remote_list.pl` - Maps between remote label and URL
+    * `FOSS_repo_list.pl` - Specifies which branches (or tags) are checked out to
+      build each component repository.
+    * `FOSS_staging_list.pl` - defines the staging order and details.
+
+## Set-up
+
+### Ubuntu 16.04
+
+The following steps assume that your are starting with a clean VM and are
+logged in as a non-root user with `sudo` privileges.
+
+    sudo apt-get update
+    sudo apt-get install software-properties-common openjdk-8-jdk ant ruby git maven build-essential
+
+### Ubuntu 14.04
+
+The following steps assume that your are starting with a clean VM and are
+logged in as a non-root user with `sudo` privileges.
+
+    sudo apt-get install software-properties-common 
+    sudo add-apt-repository ppa:openjdk-r/ppa
+    sudo apt-get update
+    sudo update-ca-certificates -f
+    sudo apt-get install openjdk-8-jdk ant ruby git maven build-essential
+
+### Ubuntu 12.04
+
+    sudo apt-get install python-software-properties software-properties-common
+    sudo add-apt-repository ppa:openjdk-r/ppa
+    sudo apt-get update
+    sudo update-ca-certificates -f
+    sudo apt-get install openjdk-8-jdk ant ruby git maven build-essential zlib1g-dev
+
+### CentOS 7
+
+    sudo yum groupinstall 'Development Tools'
+    sudo yum install java-1.8.0-openjdk ant ruby git maven cpan wget perl-IPC-Cmd
+
+### CentOS 6
+
+    sudo yum groupinstall 'Development Tools'
+    sudo yum remove java-1.7.0-openjdk java-1.6.0-openjdk ant
+    sudo yum install java-1.8.0-openjdk java-1.8.0-openjdk-devel ruby git cpan wget
+    # install specific perl modules
+    sudo cpan IPC::Cmd
+    cd /tmp
+    # install maven
+    wget http://mirror.metrocast.net/apache/maven/maven-3/3.3.9/binaries/apache-maven-3.3.9-bin.tar.gz
+    sudo tar -xf apache-maven-3.3.9-bin.tar.gz
+    sudo mv apache-maven-3.3.9 /opt
+    sudo vi /etc/profile.d/maven.sh
+    export PATH=/opt/apache-maven-3.3.9/bin:$PATH
```

Comment:
Note: we could make this a one-liner and avoid vi (same applies below):
`echo 'export PATH="/opt/apache-maven-3.3.9/bin:$PATH"' | sudo tee -a /etc/profile.d/maven.sh`